source/elements/oneDAL/source/glossary.rst

```diff
@@ -0,0 +1,170 @@
+
+Machine learning terms
+======================
+
+.. glossary::
+    :sorted:
+
+    Categorical feature
+        A :term:`feature` with a discrete domain without ordering between values.
+
+        Synonyms: nominal feature.
+
+    Classification
+        A :term:`supervised machine-learning problem<Supervised learning>` of assigning
+        discrete :term:`labels<Label>` (:term:`categorical<Categorical feature>` or :term:`ordinal<Ordinal feature>`)
+        to :term:`feature vectors<Feature vector>`.
+
+    Clustering
+        An :term:`unsupervised machine-learning problem<Unsupervised learning>` of grouping
+        :term:`feature vectors<Feature vector>` onto (usually
+        :term:`categorical<Categorical feature>`) groups without prior information on them.
+
+    Continuous feature
+        A :term:`feature` with values in a domain of real numbers.
+
+        Synonyms: quantitative feature, numerical feature.
+
+    Dataset
+        A collection of :term:`observations<Observation>`.
+
+    Feature
+        A particular property or quality of a real object or an event.
+        Has a defined type and domain.
+
+        Synonyms: attribute, variable
+
+    Feature value
+        A single point in the domain of the :term:`feature`.
+
+    Feature vector
+        A vector that encodes information about single real object or an event.
+        Contains at least one :term:`feature value`.
+
+    Inference
+        A process of applying a :term:`trained<Training>` :term:`model` to the :term:`inference set` with a goal to
+        make outputs based on input :term:`feature vectors<Feature vector>`.
+
+        Synonyms: prediction.
+
+    Inference set
+        An unlabeled :term:`dataset` used at the :term:`inference` stage to make an outputs from the :term:`model`.
+
+    Label
+        A :term:`feature` of some object or event which value is target to be predicted in :term:`supervised learning` task.
+        On the :term:`training` stage, the model which finds the best dependency between input :term:`feature vector` and
+        :term:`labels<Label>` is selected. :term:`Label` is one of outputs can be made by :term:`model`
+        on the :term:`inference` stage.
+
+    Model
+        A result of :term:`training` machine-learning algorithm that stores information necessary to run
+        :term:`inference` on a new :term:`dataset`.
+
+    Observation
+        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
+        Each of :term:`labels<Label>` and :term:`features<Feature>` can encode different objects or events.
+
+        Synonyms: instance, sample.
+
+    Ordinal feature
+        A :term:`feature` with a discrete set of possible values and some ordering between them.
+
+    Regression
+        A :term:`supervised machine-learning problem<Supervised learning>` of assigning
+        :term:`continuous<Continuous feature>` :term:`labels<Label>` to
+        :term:`feature vectors<Feature vector>`.
```

Comment:
the definition is still incomplete

Reply:
Changed `label` to `response`. Cannot see why it is incomplete. Even in the source mentioned by you, the regression task is described similar to my definition:
>We tend to refer to problems with a quantitative response as regression problems